libs/dal/src/repositories/integration/integration.repository.ts

```
@@ -12,4 +13,19 @@ export class IntegrationRepository extends BaseRepository<IntegrationEntity> {
       _applicationId: applicationId,
     });
   }
+
+  async create(data: Partial<IntegrationEntity>): Promise<IntegrationEntity> {
+    const test = await this.find({
+      _applicationId: data._applicationId,
+      providerId: data.providerId,
+      channel: data.channel,
+    });
+    if (test.length === 0) {
+      return await super.create(data);
+    }
+
+    throw new UnprocessableEntityException(
```

Here if we can throw a native error instead of importing nestjs would be great